### PR TITLE
feat: add bubble notification to sidebar nav and request tabs

### PR DESCRIPTION
**Pull request by adamstankiewicz**
Settings---edX-Admin-Portal (1)

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/Sidebar/IconLink.jsx [hunk 1]
```diff
@@ -1,51 +1,107 @@
...
+  icon,
+  isExpanded,
+  title,
+  notification,
+  ...rest
+}) => {
+  const iconRef = useRef();
+  const titleRef = useRef();
+
+  const [notificationBubbleLeft, setNotificationBubbleLeft] = useState(0);
```
Comment (long74100) [inline src/components/Sidebar/IconLink.jsx:20]: nit: notificationBubbleLeft -> notificationBubbleMarginLeft?

File: src/components/Sidebar/IconLink.jsx [hunk 2]
```diff
@@ -1,51 +1,107 @@
...
   );
 };
 
-IconLink.defaultProps = {
-  icon: null,
+const commonPropTypes = {
+  title: PropTypes.string.isRequired,
+  icon: PropTypes.shape().isRequired,
+  isExpanded: PropTypes.bool,
+  notification: PropTypes.bool,
```
Comment (long74100) [inline src/components/Sidebar/IconLink.jsx:70]: nit: hasNotification or showNotification?

File: src/components/subsidy-requests/data/hooks.js [hunk 3]
```diff
@@ -124,3 +123,62 @@ export const useSubsidyRequestConfiguration = (enterpriseUUID) => {
...
 };
+
+/**
+ * Fetches overview of subsidy requests for both subscriptions and coupon codes in order to
+ * determine counts for notification bubbles.
+ *
+ */
+export const useSubsidyRequestsOverview = (enterpriseId) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [subsidyRequestsCounts, setsubsidyRequestsCounts] = useState({
```
Comment (long74100) [inline src/components/subsidy-requests/data/hooks.js:134]: nit: setSubsidyRequestsCount

**Review by long74100 — APPROVED** [review]
just a couple nits, looks good!

File: src/components/CodeManagement/CouponCodeTabs.jsx [hunk 4]
```diff
@@ -22,11 +22,22 @@ import {
...
 
 const CouponCodeTabs = ({ enterpriseSlug }) => {
-  const { subsidyRequestConfiguration } = useContext(SubsidyRequestConfigurationContext);
+  const { subsidyRequestConfiguration, subsidyRequestsCounts } = useContext(SubsidyRequestsContext);
   const isSubsidyRequestsEnabled = subsidyRequestConfiguration?.subsidyRequestsEnabled;
   const subsidyType = subsidyRequestConfiguration?.subsidyType;
   const isRequestsTabShown = isSubsidyRequestsEnabled && subsidyType === SUPPORTED_SUBSIDY_TYPES.coupon;
 
+  let requestsTabNotification;
+  const hasRequests = subsidyRequestsCounts.couponCodes && subsidyRequestsCounts.couponCodes > 0;
```
[the remark this answers is not in the document]
Comment (adamstankiewicz) [reply src/components/CodeManagement/CouponCodeTabs.jsx:31]: I originally had the default as 0, but opted to switch it to `undefined`, but that was before I really knew what the implementation might look like consuming the hook. So, yes, I agree :)

[the remark this answers is not in the document]
Comment (adamstankiewicz) [reply src/components/CodeManagement/CouponCodeTabs.jsx:31]: Main reason why I had it as `undefined` was to differentiate between unset vs. actually having 0. But on how it ultimately gets used, I think having the default 0 makes sense.

**Pull request merged**